### PR TITLE
Frontend: com_newsfeed view review

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Frontend: com_newsfeed view review

### Testing Instructions

Install testing data and test the com_newsfeed frontend views

### Expected result

works

### Actual result

works with cs errors

### Documentation Changes Required

none

### CODE REVIEW

File: components/com_newsfeeds/views/categories/tmpl/default_items.php [hunk 1]
```diff
@@ -11,22 +11,20 @@
...
-			if (!isset($this->items[$this->parent->id][$id + 1]))
-			{
-				$class = ' class="last"';
-			}
-			?>
+		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
+			<?php if (!isset($this->items[$this->parent->id][$id + 1])) :  ?>
+				<?php $class = ' class="last"'; ?>
+			<?php endif; ?>
 			<div <?php echo $class; ?> >
```
Comment (Quy) [inline components/com_newsfeeds/views/categories/tmpl/default_items.php]: Remove space before and after.

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 2]
```diff
@@ -9,137 +9,114 @@
...
+							<div class="feed-item-description">
+								<?php if ($this->params->get('show_feed_image', 0) == 0) : ?>
+									<?php $text = JFilterOutput::stripImages($text); ?>
+								<?php endif; ?>
+								<?php $text = JHtml::_('string.truncate', $text, $this->params->get('feed_character_count')); ?>
+								<?php echo str_replace('&apos;', "'", $text); ?>
+							</div>
+						<?php endif; ?>
+					</li>
+				<?php endfor ?>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Add `;`

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 3]
```diff
@@ -9,137 +9,114 @@
...
+			<a href="<?php echo $this->item->link; ?>" target="_blank">
+				<?php echo str_replace('&apos;', "'", $this->item->name); ?>
+			</a>
+		</h2>
+		<?php if ($this->params->get('show_tags', 1)) : ?>
+			<?php $this->item->tagLayout = new JLayoutFile('joomla.content.tags'); ?>
+			<?php echo $this->item->tagLayout->render($this->item->tags->itemTags); ?>
+		<?php endif; ?>
+		<!-- Show Images from Component -->
+		<?php if (isset($images->image_first) and !empty($images->image_first)) : ?>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Change to `&&`

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 4]
```diff
@@ -9,137 +9,114 @@
...
-						$text = JFilterOutput::stripImages($text);
-					}
-					$text = JHtml::_('string.truncate', $text, $this->params->get('feed_character_count'));
-						echo str_replace('&apos;', "'", $text);
-					?>
-					</div>
+				src="<?php echo htmlspecialchars($images->image_first, ENT_COMPAT, 'UTF-8'); ?>" alt="<?php echo htmlspecialchars($images->image_first_alt, ENT_COMPAT, 'UTF-8'); ?>"/>
+			</div>
+		<?php endif; ?>
+		<?php if (isset($images->image_second) and !empty($images->image_second)) : ?>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Change to `&&`

File: components/com_newsfeeds/views/category/tmpl/default_items.php [hunk 5]
```diff
@@ -14,30 +14,28 @@
...
-					<?php echo JText::_('JGLOBAL_DISPLAY_NUM'); ?>
-				</label>
-				<?php echo $this->pagination->getLimitBox(); ?>
-			</div>
+	<form action="<?php echo htmlspecialchars(JUri::getInstance()->toString(), ENT_COMPAT, 'UTF-8'); ?>" method="post" name="adminForm" id="adminForm">
+		<?php if ($this->params->get('filter_field') !== 'hide' || $this->params->get('show_pagination_limit')) : ?>
+			<fieldset class="filters btn-toolbar">
+				<?php if ($this->params->get('filter_field') !== 'hide' && $this->params->get('filter_field') == '1') : ?>
+					<div class="btn-group">
+						<label class="filter-search-lbl element-invisible" for="filter-search"><span class="label label-warning"><?php echo JText::_('JUNPUBLISHED'); ?></span><?php echo JText::_('COM_NEWSFEEDS_FILTER_LABEL') . '&#160;'; ?></label>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default_items.php]: Split into multiple lines.

File: components/com_newsfeeds/views/category/tmpl/default_children.php [hunk 6]
```diff
@@ -9,49 +9,47 @@
...
+<?php if ($this->maxLevel != 0 && count($this->children[$this->category->id]) > 0) : ?>
+	<ul>
+		<?php foreach ($this->children[$this->category->id] as $id => $child) : ?>
+			<?php if ($this->params->get('show_empty_categories') || $child->numitems || count($child->getChildren())) : ?>
+				<?php if (!isset($this->children[$this->category->id][$id + 1])) : ?>
+					<?php $class = ' class="last"'; ?>
+				<?php endif; ?>
+				<li<?php echo $class; ?>>
+					<?php $class = ''; ?>
+					<span class="item-title"><a href="<?php echo JRoute::_(NewsfeedsHelperRoute::getCategoryRoute($child->id)); ?>">
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default_children.php]: Split into multiple lines.

File: components/com_newsfeeds/views/categories/tmpl/default_items.php [hunk 7]
```diff
@@ -11,22 +11,20 @@
...
-		if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) :
-			if (!isset($this->items[$this->parent->id][$id + 1]))
-			{
-				$class = ' class="last"';
-			}
-			?>
-			<div <?php echo $class; ?> >
-			<?php $class = ''; ?>
+		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
+			<?php if (!isset($this->items[$this->parent->id][$id + 1])) :  ?>
```
Comment (Quy) [inline components/com_newsfeeds/views/categories/tmpl/default_items.php]: Remove extra space `:  ?>`

File: components/com_newsfeeds/views/categories/tmpl/default_items.php [hunk 8]
```diff
@@ -11,22 +11,20 @@
...
+			<?php endif; ?>
+			<div<?php echo $class; ?>>
+				<?php $class = ''; ?>
 				<h3 class="page-header item-title">
-				<a href="<?php echo JRoute::_(NewsfeedsHelperRoute::getCategoryRoute($item->id, $item->language)); ?>">
-					<?php echo $this->escape($item->title); ?></a>
+					<a href="<?php echo JRoute::_(NewsfeedsHelperRoute::getCategoryRoute($item->id, $item->language)); ?>">
+						<?php echo $this->escape($item->title); ?>
+					</a>
 					<?php if ($this->params->get('show_cat_items_cat') == 1) :?>
```
Comment (Quy) [inline components/com_newsfeeds/views/categories/tmpl/default_items.php]: Add space `:?>`

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 9]
```diff
@@ -9,137 +9,114 @@
...
-					<div class="feed-item-description">
-					<?php if ($this->params->get('show_feed_image', 0) == 0)
-					{
-						$text = JFilterOutput::stripImages($text);
-					}
-					$text = JHtml::_('string.truncate', $text, $this->params->get('feed_character_count'));
-						echo str_replace('&apos;', "'", $text);
-					?>
-					</div>
+				src="<?php echo htmlspecialchars($images->image_first, ENT_COMPAT, 'UTF-8'); ?>" alt="<?php echo htmlspecialchars($images->image_first_alt, ENT_COMPAT, 'UTF-8'); ?>"/>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Add space `"/>`

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 10]
```diff
@@ -9,137 +9,114 @@
...
+		<?php if (isset($images->image_second) && !empty($images->image_second)) : ?>
+			<?php $imgfloat = empty($images->float_second) ? $this->params->get('float_second') : $images->float_second; ?>
+			<div class="pull-<?php echo htmlspecialchars($imgfloat, ENT_COMPAT, 'UTF-8'); ?> item-image">
+				<img
+				<?php if ($images->image_second_caption) : ?>
+					<?php echo 'class="caption"' . ' title="' . htmlspecialchars($images->image_second_caption) . '"'; ?>
 				<?php endif; ?>
-				</li>
-			<?php } ?>
+				src="<?php echo htmlspecialchars($images->image_second, ENT_COMPAT, 'UTF-8'); ?>" alt="<?php echo htmlspecialchars($images->image_second_alt, ENT_COMPAT, 'UTF-8'); ?>"/>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Add space `"/>`

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 11]
```diff
@@ -67,7 +67,7 @@
 				<?php if ($images->image_second_caption) : ?>
 					<?php echo 'class="caption"' . ' title="' . htmlspecialchars($images->image_second_caption) . '"'; ?>
 				<?php endif; ?>
-				src="<?php echo htmlspecialchars($images->image_second, ENT_COMPAT, 'UTF-8'); ?>" alt="<?php echo htmlspecialchars($images->image_second_alt, ENT_COMPAT, 'UTF-8'); ?>"/>
+				src="<?php echo htmlspecialchars($images->image_second, ENT_COMPAT, 'UTF-8'); ?>" alt="<?php echo htmlspecialchars($images->image_second_alt, ENT_COMPAT, 'UTF-8'); ?> "/>
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Move space after `"`

File: components/com_newsfeeds/views/category/tmpl/default.php [hunk 12]
```diff
@@ -43,9 +42,7 @@
 			<div class="clr"></div>
 		</div>
 	<?php endif; ?>
-
 	<?php echo $this->loadTemplate('items'); ?>
-
 	<?php if ($this->maxLevel != 0 && !empty($this->children[$this->category->id])) : ?>
 		<div class="cat-children">
 			<h3><?php echo JText::_('JGLOBAL_SUBCATEGORIES'); ?></h3>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default.php]: Multiple lines?

File: components/com_newsfeeds/views/category/tmpl/default.php [hunk 13]
```diff
@@ -43,9 +42,7 @@
 			<div class="clr"></div>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default.php:42]: Line 37: Add space before `/`

File: components/com_newsfeeds/views/category/tmpl/default_items.php [hunk 14]
```diff
@@ -70,10 +73,9 @@
 				</li>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default_items.php:78]: Line 49:  Remove space `" >`

Line 58-59: Multiple lines

Line 63: Multiple lines

Line 69:  Remove tab and multiple lines

**Comment by n9iels** [comment]
I have tested this item :white_check_mark: successfully on 83f3f14caa83889b8972b01b71c5c52f72261c42<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17233">issues.joomla.org/tracker/joomla-cms/17233</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 83f3f14caa83889b8972b01b71c5c52f72261c42<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17233">issues.joomla.org/tracker/joomla-cms/17233</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**